.github/workflows/cache_cleanup.yml

```diff
@@ -0,0 +1,34 @@
+name: "[🌋] Merged Branch Cache Cleanup"
+on:
+  pull_request:
+    types:
+      - closed
+  delete:
+
+jobs:
+  cleanup:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+        
+      - name: Cleanup
+        run: |
+          gh extension install actions/gh-actions-cache
+          
+          REPO=${{ github.repository }}
+          BRANCH="refs/pull/${{ github.event.pull_request.number }}/merge"
```

Comment:
this will fail on branch deletion as `github.event.pull_request.number` will not be available.